### ISSUE TITLE
Update errcheck to address false positives

### ISSUE THREAD

**Issue by Lekensteyn** [golangci/golangci-lint#1274]
It looks like the forked errcheck package (https://github.com/golangci/errcheck) is two years behind its current upstream (https://github.com/kisielk/errcheck). There are unfortunately some custom commits, so a trivial sync is not possible. What would it take to update it?

Is it possible to upstream some changes and avoid the use of a fork to ensure easy upgrades? (@kisielk @jirfag ?)

I ran into a false-positive `errcheck` warning with the following code:
```go
package main

import (
	"crypto/sha256"
	"fmt"
)

func main() {
	h := sha256.New()
	h.Write([]byte("food"))
	fmt.Println(h.Sum(nil))
}
```

Using the `golangci/golangci-lint:v1.27.0` Docker image it outputs:
```
main.go:10:9: Error return value of `h.Write` is not checked (errcheck)
	h.Write([]byte("food"))
	       ^
```
This issue has been solved upstream two years ago via https://github.com/kisielk/errcheck/pull/137.

Related issues:
-  https://github.com/golangci/golangci-lint/issues/1194
- Attempt at updating the fork: https://github.com/golangci/errcheck/pull/4

**Comment by SVilgelm** [comment]
I asked @kisielk here https://github.com/kisielk/errcheck/issues/184 about adding a special code for golangci, so we can use the main repo instead of fork
